### PR TITLE
[roku] Improve TV discovery model name and add timeout

### PR DESCRIPTION
**Pull request by mlobstein**
Improve the label that is given to the thing by the discovery process. Previously TV things were not easily identified as a Roku device by the model name provided by the API. The `friendly-model-name` and `user-device-location` fields are now combined to help the user identify the thing in case of duplicate devices in different rooms.  The strings are formatted to remove any extraneous characters as seen in the friendly-model-name example. 

image

Addresses #15797 by setting a timeout on http requests sent to the roku device. Previously if the device was to crash while an http request was in progress it could cause the binding to hang.

Addresses #16164 by setting the uuid property to prevent manually added things from appearing in the inbox.

### CODE REVIEW

**Review by lsiepel — APPROVED** [review]
LGTM

**Comment by morph166955** [comment]
It may be better to use friendly-device-name or default-device-name to avoid duplicates and/or make it more easily identified in the inbox.  Example:

<friendly-device-name>50" onn. Roku TV Guest Room</friendly-device-name>
<friendly-model-name>onn. Roku TV</friendly-model-name>
<default-device-name>onn. Roku TV - X01200ECK0EW</default-device-name>

<friendly-device-name>50" onn. Roku TV Girls</friendly-device-name>
<friendly-model-name>onn. Roku TV</friendly-model-name>
<default-device-name>onn. Roku TV - X01200G883DJ</default-device-name>

**Comment by mlobstein** [comment]
> It may be better to use friendly-device-name or default-device-name to avoid duplicates and/or make it more easily identified in the inbox. 

It seems redundant because the serial number (as thing id) is seen below the thing label: 
image

**Comment by morph166955** [comment]
That's fair.  The friendly name then may make the most sense as it seems to be tied to what the user set when they configured the device.  I have multiple of the same model in the house so this would make it obvious which is which.

**Comment by morph166955** [comment]
Sorry I didn't see your edit about the quotes.  Perhaps concatenate in user-device-location?  Just trying to figure out some way to make it easily identifiable.  I have 6 of the same devices in the house, having something to go "oh that's that one" is very helpful.

`<user-device-location>Guest bedroom</user-device-location>`

**Comment by mlobstein** [comment]
>  I have 6 of the same devices in the house, having something to go "oh that's that one" is very helpful.
> 
> `<user-device-location>Guest bedroom</user-device-location>`

What about `friendly-model-name` - `user-device-location`? I think that would accomplish  "oh that's that one"

**Review by morph166955 — APPROVED** [review]
LGTM and Thank You!

File: bundles/org.openhab.binding.roku/README.md [hunk 1]
```diff
@@ -109,17 +109,19 @@ roku:roku_tv:mytv1 "My Roku TV" [ hostName="192.168.10.1", refresh=10 ]
 
 String Player_ActiveApp        "Current App: [%s]"         { channel="roku:roku_player:myplayer1:activeApp" }
 String Player_ActiveAppName    "Current App Name: [%s]"    { channel="roku:roku_player:myplayer1:activeAppName" }
-String Player_Button           "Send Command to Roku"      { channel="roku:roku_player:myplayer1:button" }
+String Player_Button           "Send Command to Roku"      { channel="roku:roku_player:myplayer1:button", autoupdate="false" }
```
Comment (jlaur) [inline bundles/org.openhab.binding.roku/README.md:112]: Seeing this reminded me of #12141. Do you think a similar solution would be relevant here as well? Not for this PR obviously, just a side note.

**Review by jlaur — APPROVED** [review]
LGTM, thanks.

File: bundles/org.openhab.binding.roku/README.md (hunk 1 above)
Comment (mlobstein) [reply bundles/org.openhab.binding.roku/README.md:112]: Is that just a substitute for  

> Seeing this reminded me of #12141. Do you think a similar solution would be relevant here as well? Not for this PR obviously, just a side note.

I can address that in a future PR. Is this just a substitute/implicit way of setting autoupdate="false"?

Comment (jlaur) [reply bundles/org.openhab.binding.roku/README.md:112]: The **veto** policy will make sure that the command is sent, but without automatically updating the item state:
https://www.openhab.org/docs/developer/bindings/thing-xml.html#auto-update-policies

**Pull request merged**